packages/cli/src/commands/translate/create.ts

```diff
@@ -0,0 +1,94 @@
+import { existsSync } from 'node:fs';
+import fs from 'node:fs/promises';
+import path from 'node:path';
+
+import { isLanguageTag, type LanguageTag } from '@logto/language-kit';
+import { isBuiltInLanguageTag as isPhrasesBuiltInLanguageTag } from '@logto/phrases';
+import { isBuiltInLanguageTag as isPhrasesUiBuiltInLanguageTag } from '@logto/phrases-ui';
+import { conditionalString, type Optional } from '@silverhand/essentials';
+import type { CommandModule } from 'yargs';
+
+import { log } from '../../utils.js';
+import { inquireInstancePath } from '../connector/utils.js';
+
+import { createOpenaiApi, translate } from './openai.js';
+import { baseLanguage, readBaseLocaleFiles } from './utils.js';
+
+const createFullTranslation = async (
+  instancePath: Optional<string>,
+  packageName: 'phrases' | 'phrases-ui',
+  languageTag: LanguageTag
+) => {
+  const directory = path.join(
+    await inquireInstancePath(instancePath),
+    'packages',
+    packageName,
+    'src/locales'
+  );
+  const files = await readBaseLocaleFiles(directory);
+
+  log.info(
+    'Found ' +
+      String(files.length) +
+      ' file' +
+      conditionalString(files.length !== 1 && 's') +
+      ' in ' +
+      packageName +
+      ' to translate'
+  );
+
+  const openai = createOpenaiApi();
+
+  /* eslint-disable no-await-in-loop */
+  for (const file of files) {
+    const relativePath = path.relative(path.join(directory, baseLanguage), file);
+    const targetPath = path.join(directory, languageTag, relativePath);
+
+    if (existsSync(targetPath)) {
+      log.info(`Target path ${targetPath} exists, skipping`);
+      continue;
+    }
+
+    log.info(`Translating ${file}`);
+    const result = await translate(openai, languageTag, file);
```

Comment:
nit: feels like could create an openai requester within the translate method.

Reply:
thought about that, but it will be one instance per request, looks like not efficient. we can refactor later with class